edalize/design_compiler.py

```diff
@@ -0,0 +1,152 @@
+# Copyright edalize contributors
+# Licensed under the 2-Clause BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-2-Clause
+
+import logging
+import os.path
+import platform
+import re
+import subprocess
+
+from edalize.edatool import Edatool
+from edalize.yosys import Yosys
+from importlib import import_module
+
+logger = logging.getLogger(__name__)
+
+""" design-compiler Backend
+
+A core (usually the system core) can add the following files:
+
+- Standard design sources
+
+- Libraries
+"""
+class Design_compiler(Edatool):
+
+    argtypes = ['vlogdefine', 'vlogparam', 'generic']
+
+    @classmethod
+    def get_doc(cls, api_ver):
+        if api_ver == 0:
+            return {'description' : "The design_compiler backend executes Synopsys design_copiler to build a gate-level netlist",
+                    'members' : [
+                        {'name' : 'script_dir',
+                         'type' : 'String',
+                         'desc' : 'Path to Syopsys scripts (e.g. /home/user/project/synopsys/scripts)'},
+                         {'name' : 'dc_script',
```

Comment:
Perhaps this would be better as either a tclSource file or a "new filetype" along with the main sources?

Reply:
you mean replacing "String" with tclFile?

Reply:
No, he means using the tclSource file type in the core file and then copying any accompanying files over.  
 E.g.: https://github.com/lowRISC/opentitan/blob/master/hw/top_earlgrey/chip_earlgrey_cw310.core

Reply:
correct me if I am wrong, but wouldn't this assume that there is a kind of ordering of calls to those scripts?

calling a "root" script like it is now, allows people to define their flow in that script, and get rid of the boring, lengthy, and error-prone "analyze" script to compile all the files. This one is very well done by FuseSoc.

But making a synthesis flow one-fit-all does not sound very versatile to me.

But you reminded me that actually, it makes sense to support the tcl files by sourcing them in the analyze script.

Of course, if you have suggestions I would be happy to work on them.

As today the Makefile only calls the "root" script, and then from that script you call all the rest, that's why the TCL file was not actually included in the analyze script. But why not. Let me do it for completeness reasons.

Reply:
done in next commit @shareefj @Jbalkind